src/engine/TPA.cc

```diff
@@ -982,16 +983,101 @@ bool TPASplit::verifyExactPower(unsigned short power) const {
     return res == s_False;
 }
 
+vec<PTRef> TPABase::houdiniCheck(PTRef invCandidates, PTRef transition, SafetyExplanation::FixedPointType allignment) {
+    SMTConfig config;
+    MainSolver solver(logic, config, "Fixed-point checker");
+    solver.push();
+    auto candidates = topLevelConjuncts(logic, invCandidates);
+    if (allignment == SafetyExplanation::FixedPointType::RIGHT) {
+        solver.insertFormula(getNextVersion(transition));
+    } else {
+        if (allignment == SafetyExplanation::FixedPointType::LEFT) { solver.insertFormula(transition); }
+    }
+    solver.push();
+
+    // RIGHT:
+    //   rightInvariants /\ currentLevelTransition /\ getNextVersion(transition) =>
+    //     shiftOnlyNextVars(currentLevelTransition);
+    // LEFT:
+    //   leftInvariants /\ transition /\ getNextVersion(currentLevelTransition) =>
+    //     shiftOnlyNextVars(currentLevelTransition);
+    while (candidates.size() > 128) {
+        for (int i = candidates.size() - 1; i >= 1; i -= 2) {
+            PTRef n_f = logic.mkAnd(candidates[i], candidates[i - 1]);
+            candidates.pop();
+            candidates.pop();
+            candidates.push(n_f);
+            if (candidates.size() <= 128) { break; }
+        }
+    }
```

Comment:
I think this is not doing what it was intended to do.
I tried to simulate what it is doing and I believe this is going to basically end up removing all elements after the cutoff (128), instead of combining them into a single conjunction.

Reply:
Please refactor this loop to a separate method.